### PR TITLE
refactor: heroes

### PR DESCRIPTION
**Pull request by kristofferlarberg**
# Description
This pr refactors the different hero components on the site, resulting in one reusable component (`BaseHero`) which is used in the common hero component and the one specifically for `/iteamX`. The specific hero components for coworker and start page respectively are replaced by the common hero component.

## Reminder
Run `sanity deploy` from branch + edit content in CMS before merge.

## Visual change
The start page now has the same gradient overlay as other pages. It's a bit darker, but also more legible (approved by UX).
Skärmavbild 2022-10-27 kl  10 49 56

### CODE REVIEW

File: client/src/components/Header/Header.js [hunk 1]
```diff
@@ -197,7 +197,7 @@ export const Header = ({ isDark = false, loaded }) => {
   const [open, setOpen] = useState(false)
 
   const size = useWindowSize()
-  const isMobile = size.width < 622
+  const narrowScreen = size.width < 1024
```
Comment (mull) [inline client/src/components/Header/Header.js:200]: Would be nice to move this to a constant somewhere. Do we have other magic numbers in the code?

**Review by mull — APPROVED** [review]

File: client/src/components/Header/Header.js [hunk 2]
```diff
@@ -267,7 +267,7 @@ export const Header = ({ isDark = false, loaded }) => {
             <nav className="navbar site-navbar offcanvas-active navbar-expand-lg navbar-light">
               {/* <!-- Brand Logo--> */}
               <div className="brand-logo">
-                {isMobile ? (
+                {narrowScreen ? (
```
Comment (momentiris) [inline client/src/components/Header/Header.js:270]: Perhaps not a part of this PR, but I'd say just remove that `narrowScreen`stuff. Tailwind already supports this. So we could just show/hide those element that uses this variable with css

File: client/src/features/Hero/BaseHero.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    <SectionWithBackground
+      mediaType={mediaType}
+      className={clsx(
+        className,
+        'tw-flex tw-flex-col tw-justify-center tw-pt-32 tw-pb-14 lg:tw-pt-48 lg:tw-pb-48'
+      )}
+    >
+      <Container className="tw-h-full">
+        <Row>
+          <Col md="11" lg="9" xl="7">
```
Comment (momentiris) [inline client/src/features/Hero/BaseHero.js:72]: Maybe take the opportunity to swap out bootstrap here while at it?

File: client/src/features/IteamX/Hero.js [hunk 4]
```diff
@@ -1,52 +1,16 @@
...
 
-import { Section, Box, Typography, Span, CTALink } from '../../components/Core'
-import { getExternalOrInternalLink } from '../../utils/helpers'
+import { Typography, Span } from '../../components/Core'
+import { BaseHero } from '../Hero'
 
-export const Hero = ({ content }) => {
-  const [title] = content.title.split('.')
+export const Hero = ({ title, subtitle, mediaType, link }) => {
+  const splitTitle = title.split('.')
```
Comment (momentiris) [inline client/src/features/IteamX/Hero.js:7]: What does this do?
looks like `splitTitle` is now an array.

File: client/src/pages/about/[slug].js [hunk 5]
```diff
@@ -15,21 +16,19 @@ const CoworkerPage = ({ data, preview = false }) => {
...
-          >
-            {post?.role ?? post.role}
-          </CoworkerHero>
-          <Medarbetare info={post ?? post} />
-        </>
+        <Hero
+          title={post.fullname && post.fullname}
+          subtitle={post.role && post.role}
+          mediaType={post.heroImage && customMediaType}
+        />
```
Comment (momentiris) [inline client/src/pages/about/[slug].js:29]:
```suggestion
        <Hero
          title={post.fullname}
          subtitle={post.role}
          mediaType={post.heroImage && customMediaType}
        />
```
`post.fullname && post.fullname` will reteurn the same value as `post.fullname`

File: client/src/pages/about/[slug].js [hunk 6]
```diff
@@ -15,21 +16,19 @@ const CoworkerPage = ({ data, preview = false }) => {
...
       )}
+      <Medarbetare info={post ?? post} />
```
Comment (momentiris) [inline client/src/pages/about/[slug].js:31]:
```suggestion
      <Medarbetare info={post} />
```
`post ?? post´ this basically says "post or post"

File: client/src/pages/about/index.js [hunk 7]
```diff
@@ -48,7 +49,14 @@ const About = ({ data, preview = false }) => {
             }}
           />
         )}
-        <Hero content={hero && hero} />
+        {hero && (
+          <Hero
+            title={hero.title && hero.title}
+            subtitle={hero.subtitle && hero.subtitle}
+            mediaType={hero.mediaType && hero.mediaType}
+            link={hero.link && hero.link}
```
Comment (momentiris) [inline client/src/pages/about/index.js:57]:
```suggestion
            title={hero.title}
            subtitle={hero.subtitle}
            mediaType={hero.mediaType}
            link={hero.link}
```

Same here. If we want to defend ourselvs against undefined, we'll have to do that inside the `<Hero />`. Alternatively not renderingn the hero at all if those values does not exist

File: client/src/pages/book.js [hunk 8]
```diff
@@ -62,7 +64,14 @@ const Book = ({ data, preview = false }) => {
...
         )}
         {preview && <ExitPreviewLink />}
-        <Hero content={page?.hero && page.hero} />
+        {hero && (
+          <Hero
+            title={hero.title && hero.title}
+            subtitle={hero.subtitle && hero.subtitle}
+            mediaType={hero.mediaType && hero.mediaType}
+            link={hero.link && hero.link}
+          />
```
Comment (momentiris) [inline client/src/pages/book.js:73]: same here

File: client/src/pages/index.js [hunk 9]
```diff
@@ -63,7 +63,15 @@ const StartPage = ({ data, preview = false }) => {
...
-      {page?.hero && <Hero content={page.hero} />}
+      {hero && (
+        <Hero
+          title={hero.title && hero.title}
+          subtitle={hero.subtitle && hero.subtitle}
+          mediaType={hero.mediaType && hero.mediaType}
+          link={hero.link && hero.link}
+          className="tw-h-screen"
+        />
+      )}
```
Comment (momentiris) [inline client/src/pages/index.js:74]: +1

File: client/src/pages/iteamX.js [hunk 10]
```diff
@@ -55,7 +57,14 @@ const IteamX = ({ data, preview = false }) => {
...
         )}
-        {xPage.hero && <Hero content={xPage.hero} />}
+        {hero && (
+          <Hero
+            title={hero.title && hero.title}
+            subtitle={hero.subtitle && hero.subtitle}
+            mediaType={hero.mediaType && hero.mediaType}
+            link={hero.link && hero.link}
+          />
+        )}
```
Comment (momentiris) [inline client/src/pages/iteamX.js:67]: +1

File: client/src/pages/karriar/index.js [hunk 11]
```diff
@@ -54,7 +54,14 @@ const Career = ({ data, preview = false, carousel }) => {
...
         {preview && <ExitPreviewLink />}
-        <Hero content={hero} />
+        {hero && (
+          <Hero
+            title={hero.title && hero.title}
+            subtitle={hero.subtitle && hero.subtitle}
+            mediaType={hero.mediaType && hero.mediaType}
+            link={hero.link && hero.link}
+          />
+        )}
```
Comment (momentiris) [inline client/src/pages/karriar/index.js:64]: +1

File: client/src/pages/mvp.js [hunk 12]
```diff
@@ -50,7 +53,14 @@ const Mvp = ({ data, preview = false }) => {
...
       )}
-      <Hero content={mvpPage?.hero && mvpPage.hero} />
+      {hero && (
+        <Hero
+          title={hero.title && hero.title}
+          subtitle={hero.subtitle && hero.subtitle}
+          mediaType={hero.mediaType && hero.mediaType}
+          link={hero.link && hero.link}
+        />
+      )}
```
Comment (momentiris) [inline client/src/pages/mvp.js:63]: +1

File: client/src/pages/scaleup.js [hunk 13]
```diff
@@ -56,7 +58,14 @@ const ScaleUp = ({ data, preview = false, carousel, ourPricing }) => {
...
       )}
-      {scaleUpPage?.hero && <Hero content={scaleUpPage.hero} />}
+      {hero && (
+        <Hero
+          title={hero.title && hero.title}
+          subtitle={hero.subtitle && hero.subtitle}
+          mediaType={hero.mediaType && hero.mediaType}
+          link={hero.link && hero.link}
+        />
+      )}
```
Comment (momentiris) [inline client/src/pages/scaleup.js:68]: +1

File: client/src/components/Header/Header.js (hunk 1 above)
Comment (kristofferlarberg) [reply client/src/components/Header/Header.js:200]: Seems like this is the only one. I'll remove and apply condition via tailwind as @momentiris suggests below.

File: client/src/features/IteamX/Hero.js (hunk 4 above)
Comment (kristofferlarberg) [reply client/src/features/IteamX/Hero.js:7]: Yes, strange how it worked anyway. I'll add the array destructuring again.

File: client/src/pages/about/[slug].js (hunk 5 above)
Comment (kristofferlarberg) [reply client/src/pages/about/[slug].js:29]: Hmm yes, these existed from before but I have been unsure about it. I'll move the checks to inside the component.

**Pull request merged**